src/main/python/pybuilder/pip_utils.py

```diff
@@ -41,10 +41,69 @@ def build_dependency_version_string(mixed):
     return version
 
 
-def pip_install(install_targets, index_url=None, extra_index_url=None, upgrade=False, insecure_installs=None,
-                force_reinstall=False, target_dir=None, verbose=False, trusted_host=None, constraint_file=None,
+def pip_install_get_output(install_targets, index_url=None,
+                           extra_index_url=None, upgrade=False,
+                           insecure_installs=None,
+                           force_reinstall=False, target_dir=None,
+                           verbose=False, trusted_host=None,
+                           constraint_file=None, eager_upgrade=False,
+                           logger=None, env=None, cwd=None):
+    command_line = _create_pip_command_line(install_targets=install_targets,
+                                            index_url=index_url,
+                                            extra_index_url=extra_index_url,
+                                            upgrade=upgrade,
+                                            insecure_installs=insecure_installs,
+                                            force_reinstall=force_reinstall,
+                                            target_dir=target_dir,
+                                            verbose=verbose,
+                                            trusted_host=trusted_host,
+                                            constraint_file=constraint_file,
+                                            eager_upgrade=eager_upgrade)
+    if env is None:
+        env = os.environ
+
+    if logger:
+        logger.debug("Invoking pip: %s", command_line)
+    return execute_commandstr_and_capture_output(command_line, env=env, cwd=cwd)
+
+
+def pip_install(install_targets, index_url=None, extra_index_url=None,
```

Comment:
This can be de-duplicated, just call the above `pip_intsall_get_output` and then drop the output instead of returning it.